schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+publish it.
+
+For formats that do not depend on external schemas for serialization, schemas
+might still be required in some scenarios to allow for consumers or inspecting
+intermediaries to validate the data structure's compliance with a set of rules.
+
+This specification defines a very simple API focused on storing, organizing, and
+accessing such documents. While there are very sophisticated metadata management
+solutions available as open source und under neutral governance already, the focus
+of this specification is not great sophistication but straightforward simplicity.
+
+This schema registry deals with only three top-level elements:
+
+- Schema Group: A named collection of schemas. Each group holds a logically
+  related set of schemas, typically belonging to a particular application or
+  subject to shared access control rules.
+- Schema: A document describing the structure, names, and types of some
+  structured data payload.
+- Version: A specific version of a schema document. Even though not prescribed
+  in this specification, an implementation might choose to impose compatibility
+  constraints on versions following the initial version of a schema.
+
+> NOTE: There are two fundamentally different options for how we can organize
+> multiple schema formats and this specification draft reflects both: 
+>
+> 1) In an "ideal" REST service, any data entity can be represented ("R" in
+>    REST) in multiple different data formats, which a client can request in
+>    prioritized order during content-type negotiation. If a service offers up
+>    the same entity in JSON, Protobuf, and Avro formats, there would therefore
+>    be three distinct schemas, each defining a representation of the data
+>    entity for the respective format. From the same "REST" mindset perspective,
+>    the ideal representation of this set of schemas, which differ with regards
+>    to the format they describe, but ought to be semantically identical, is to
+>    also differentiate them only by their content type and maintain them under
+>    the exact same name as if they were one schema.
+> 2) Since the above strategy is truly RESTful, but quite esoteric if you've not
+>    grown up as a RESTafarian, the alternative strategy for concurrently
+>    handling multiple schema formats is much simpler: Constrain each schema
+>    group to a single format.
+
+## 2. Schema Registry Elements
+
+This section further describes the elements enumerated in the introduction.
+
+### 2.1. Schema Group
+
+A schema group is a named collection of schemas. Since a schema registry is
+often a resource with a scope greater than a single application and might even
+span multiple organizations, it is very useful to put a grouping construct
+around sets of schemas that are related either by ownership or by a shared
+subject matter context.
+
+Implementations of this specification MAY associate access control rules with
```

Comment:
Do you see the spec talking about access control at levels below Schema Group too? If so, I wonder if this MAY/sentence should be in a more global spot to talk about access control at any layer in the hierarchy

Reply:
We can surely open that up for also having access control levers at the other levels.